ac.rkt

```diff
@@ -644,18 +768,32 @@
 ;   and it's bound to a function, generate (foo bar) instead of
 ;   (ar-funcall1 foo bar)
 
+(define (stx? expr)
+  (and (symbol? expr) (eqv? (bound? expr) 'syntax)))
+
 (define (ac-call fn args env)
+  (if (not (eqv? (lang*) 'arc))
+      `(,(ac fn env) ,@(map (lambda (x) (ac x env)) args))
   (let ([macfn (ac-macro? fn)])
     (cond [macfn
            (ac-mac-call macfn args env)]
+          [(stx? fn)
+           (parameterize ((lang* 'rkt))
+             `(,(ac fn env) ,@(map (lambda (x) (ac x env)) args)))]
+          ((and (id-literal? fn)
+                (void? (id-literal fn)))
+           (map (lambda (x) (ac x env)) args))
```

Comment:
This seems incorrect. The point of `ac` is to compile an Arc expression into a Racket expression. When the thing we're calling is a Racket macro, not all the subforms are necessarily supposed to be expressions; some are patterns or variable bindings or `require` specifications, etc.

By treating them as expressions, you're changing the meaning of `ac` so that sometimes it processes things that aren't really expressions. So `ac` ends up having to be a compiler and an s-expression template DSL at the same time, in different contexts.

I think Anarki's existing approach, using `$` to get into Racket and `unquote` to return to Arc, is already a better s-expression template DSL design than this is.